### PR TITLE
Improve spar test coverage

### PR DESCRIPTION
**Pull request by fisx**
https://wearezeta.atlassian.net/browse/WPB-5708

## Checklist

 - [x] Add a new entry in an appropriate subdirectory of `changelog.d`
 - [x] Read and follow the [PR guidelines](https://docs.wire.com/developer/developer/pr-guidelines.html)

### CODE REVIEW

File: services/spar/test-integration/Test/Spar/APISpec.hs [hunk 1]
```diff
@@ -1058,44 +1061,70 @@ specCRUDIdentityProvider = do
...
+      $ \(provisionViaScim, updateNotReplace, externalIdIsEmail) -> do
+        it ("creates new idp, setting old_issuer; sets replaced_by in old idp; scim user search still works: provisionViaScim=" <> show provisionViaScim <> ", updateNotReplace=" <> show updateNotReplace <> ", externalIdIsEmail=" <> show externalIdIsEmail) $ do
           env <- ask
-          (owner1, teamid, idp1, (IdPMetadataValue _ idpmeta1, _privCreds)) <- registerTestIdPWithMeta
+          (owner1, teamid, idp1, (IdPMetadataValue _ idpmeta1, privcreds)) <- registerTestIdPWithMeta
           let idp1id = idp1 ^. idpId
 
-          mbScimStuff :: Maybe (ScimToken, Scim.StoredUser SparTag, Scim.User SparTag) <-
-            if haveScim
+          (tok, userStuff) :: (ScimToken, Either (UserId, Maybe Text, Text) (Scim.StoredUser SparTag, Scim.User SparTag)) <- do
```
Comment (akshaymankar) [inline services/spar/test-integration/Test/Spar/APISpec.hs:1076]: What does `userStuff` mean, can we have better name for this thing? The type really doesn't help. At this point I am questioning why not just write separate tests for Scim and without scim?

**Review by akshaymankar — COMMENTED** [review]
This test is so hard to read. I have two ideas:
1. Split the tests for with scim and without scim to separate things instead of `forM_` over this.
2. Move this test to the new test framework as this is what we had decided anyway, i.e. whenever we touch a test in old suite, we move it to the new one.

**Review by akshaymankar — APPROVED** [review]

File: services/spar/test-integration/Test/Spar/APISpec.hs [hunk 2]
```diff
@@ -1058,35 +1061,80 @@ specCRUDIdentityProvider = do
...
+                let erase :: IdP -> IdP
+                    erase =
+                      (idpId .~ (idp1 ^. idpId))
+                        . (idpMetadata . edIssuer .~ (idp1 ^. idpMetadata . edIssuer))
+                        . (idpExtraInfo . oldIssuers .~ (idp1 ^. idpExtraInfo . oldIssuers))
+                        . (idpExtraInfo . replacedBy .~ (idp1 ^. idpExtraInfo . replacedBy))
+                        . (idpExtraInfo . handle .~ (idp1 ^. idpExtraInfo . handle))
+                 in erase idp1 `shouldBe` erase idp2
+
+        -- scim doesn't not work with more than one idp, so we can't test the post variant
```
[the remark this answers is not in the document]
Comment (fisx) [reply services/spar/test-integration/Test/Spar/APISpec.hs:1119]: i don't see the difference in your change, but i think i know what you wanted to do here.  :)  commit coming up!

replace-idp requires more than one idp to be present in the team, and scim doesn't allow that.  so until we implement that missing feature we can't test it.  we can only test replace (aka post) + saml-provisioning.

[the remark this answers is not in the document]
Comment (fisx) [reply services/spar/test-integration/Test/Spar/APISpec.hs]: ah, you just meant the typo.  ignore my explanations then.

[the remark this answers is not in the document]
Comment (akshaymankar) [reply services/spar/test-integration/Test/Spar/APISpec.hs]: Sorry I didn't mean to create a suggestion.

**Pull request merged**